compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -904,14 +906,15 @@ pub trait PrettyPrinter<'tcx>:
                     first = false;
                 }
 
-                for (assoc_item_def_id, ty) in assoc_items {
+                for (assoc_item_def_id, term) in assoc_items {
+                    let ty = if let Term::Ty(ty) = term.skip_binder() { ty } else { continue };
```

Comment:
I think we should print them instead of skipping them. if it's not a type argument, then you can skip the generator return check and just print the constant.